crates/sui-graphql-rpc/src/data/package_resolver.rs

```diff
@@ -59,9 +60,13 @@ impl Loader<PackageKey> for Db {
         let ids: BTreeSet<_> = keys.iter().map(|PackageKey(id)| id.to_vec()).collect();
         let stored_packages: Vec<StoredPackage> = self
             .execute(move |conn| {
-                conn.results(move || {
-                    dsl::packages.filter(dsl::package_id.eq_any(ids.iter().cloned()))
-                })
+                async move {
+                    conn.results(move || {
+                        dsl::packages.filter(dsl::package_id.eq_any(ids.iter().cloned()))
+                    })
+                    .await
+                }
+                .scope_boxed()
             })
```

Comment:
Would it be possible to have the transactional API `execute`, `execute_repeatable` etc accept a callback that it can call `scope_boxed` on? so that this could become:

```rust
            .execute(|conn| async move {
                conn.results(move || {
                    dsl::packages.filter(dsl::package_id.eq_any(ids.iter().cloned()))
                })
                .await
            })
```

If we can't elide it like that, I'm generally curious to learn more about `scope_boxed` and when we should and shouldn't use it.

Reply:
I tried for a while to not require this as a part of the trait api but kept hitting lifetime issues bounding the lifetime of the produced future from the provided closure in order to meet the needs of the transaction api `diesel-async` required. Once i finish this migration i'll take another pass back at this to see if it can be simplified and not require the explicit bounding with scope_boxed